packages/loopring_v3/contracts/impl/libexchange/ExchangeData.sol

```diff
@@ -168,24 +168,24 @@ library ExchangeData
         return 21888242871839275222246405745257275088548364400416034343698204186575808495617;
     }
 
-    function MAX_PROOF_GENERATION_TIME_IN_SECONDS() internal pure returns (uint32) { return 1 hours; }
-    function MAX_GAP_BETWEEN_FINALIZED_AND_VERIFIED_BLOCKS() internal pure returns (uint32) { return 2500; }
+    function MAX_PROOF_GENERATION_TIME_IN_SECONDS() internal pure returns (uint32) { return 14 days; }
+    function MAX_GAP_BETWEEN_FINALIZED_AND_VERIFIED_BLOCKS() internal pure returns (uint32) { return 1000; }
     function MAX_OPEN_DEPOSIT_REQUESTS() internal pure returns (uint16) { return 1024; }
     function MAX_OPEN_WITHDRAWAL_REQUESTS() internal pure returns (uint16) { return 1024; }
-    function MAX_AGE_UNFINALIZED_BLOCK_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 1 days; }
-    function MAX_AGE_REQUEST_UNTIL_FORCED() internal pure returns (uint32) { return 15 minutes; }
-    function MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 1 days; }
-    function MAX_TIME_IN_SHUTDOWN_BASE() internal pure returns (uint32) { return 1 days; }
-    function MAX_TIME_IN_SHUTDOWN_DELTA() internal pure returns (uint32) { return 15 seconds; }
-    function TIMESTAMP_HALF_WINDOW_SIZE_IN_SECONDS() internal pure returns (uint32) { return 10 minutes; }
+    function MAX_AGE_UNFINALIZED_BLOCK_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 21 days; }
+    function MAX_AGE_REQUEST_UNTIL_FORCED() internal pure returns (uint32) { return 14 days; }
+    function MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 14 days; }
```

Comment:
A large value for `MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE()` allows user funds to be stuck in the exchange for a very long time because on-chain withdrawal requests only need to be handled _very_ slowly by the operator. And unlike some other constant values it doesn't cost the operator anything if he does this. So if possible this value should be lowered as soon as possible in future versions.

Reply:
I agree. The only concern with a smaller value I have is that when the data center is down and all off-chain data is lost, the DEX needs enough time to restore the Merkle tree from onchain data and bring all systems back online. This is certainly not the operator's fault, so the DEX shall not take a loss. Also, keep in mind the maintenance mode will still require operator to take care of pending requests.

Reply:
@Brechtpd The protocol's security guarantees are for the worst cases, the DEX's quality of service cannot rely on the protocol's "bottom lines".  If a DEX always handles deposits/withdrawal slowly, users will opt-out.